test/e2e/azure_privatecluster.go

```diff
@@ -134,7 +134,7 @@ func AzurePrivateClusterSpec(ctx context.Context, inputGetter func() AzurePrivat
 	result := &clusterctl.ApplyClusterTemplateAndWaitResult{}
 
 	// NOTE: We don't add control plane waiters here because Helm install will fail since the apiserver is private and not reachable from the prow cluster.
-	// As a workaround, we use still ClusterResourceSet to install CNI on the private cluster until a Helm integration is available.
+	// As a workaround, we use in-tree cloud-provider-azure and ClusterResourceSet to install CNI on the private cluster until a Helm integration is available.
```

Comment:
@Jont828 this would be a good one to test the Helm addon provider once you're ready, private clusters are essentially blocked from using out of tree cloud provider without it since we can't run the Helm install from outside the vnet.